### ISSUE TITLE
Bug with positions of objects in tilted slits

### ISSUE THREAD

**Issue by jhod0** [pypeit/PypeIt#1551]
Per @jhennawi 's request I'm moving this from a slack thread to a GitHub issue.

We found a subtle error in the way PypeIt determines the positions of objects in tilted slits. (To be fair, it is definitely subtle, and it only popped up because the DEIMOS mask design I used had a bit of a pathological edge case).

Our mask was aligned at PA=3deg with most slits along that angle, except our main target was in a tilted slit at PA=-15deg. In the image below, a typical slit at PA=3 is seen on the left where PypeIt correctly gets the position of a serendipitous object. On the right is our main target, with boxes overlaid where I manually extracted spectra. For these, PypeIt *flips* the reported positions, with spectra extracted on the right side of the slit reported on the left.

![image](https://user-images.githubusercontent.com/8944001/222223247-d0283a8f-d8eb-462a-b775-b838aff9d3db.png)

The reason for this seems to be that our mask design reported the -15deg slit as 165deg. This should be fine, it's "correct", but when PypeIt goes to compute the position, the 180deg shift makes the offset *backwards*.

The relevant code seems to be here:

https://github.com/pypeit/PypeIt/blob/149cf4b5c9872585a0537e721c59c7a970c0771c/pypeit/slittrace.py#L1083-L1093

I think a fix could be to see if the slit PA is >90deg different from the overall mask PA, and then adjust:

```
if obj_slit_pa[sidx] - mask_pa > 90:
    effective_slit_pa = obj_slit_pa[sidx] - 180
elif obj_slit_pa[sidx] - mask_pa < -90:
    effective_slit_pa = obj_slit_pa[sidx] + 180
else:
    effective_slit_pa = obj_slit_pa[sidx]
```
This would need some work to ensure it gets edge cases, like wrapping around 360, correct.

**Comment by profxj** [comment]
@jhod0 -- can you upload your data somewhere and provide a link?

**Comment by jhod0** [comment]
I just posted a link for Debora in the PypeIt slack!

**Comment by debora-pe** [comment]
Addressed in PR #1563.
Closing
